playbooks/operationalizing/secure-forward-splunk.adoc

```diff
@@ -0,0 +1,318 @@
+
+== Architecture
+
+As part of the aggregated logging framework within OpenShift, containerized instances of Fluentd are deployed to OpenShift nodes as link:https://docs.openshift.com/container-platform/latest/dev_guide/daemonsets.html[DaemonSets]. As messages are collected, Fluentd communicates with ElasticSearch to persistently store messages for later retrieval. Cluster administrators and users can browse, search and view stored messages using Kibana, a web based user interface.
+
+To support forwarding messages to Splunk that are captured by the aggregated logging framework, Fluentd can be configured to make use of the secure forward output plugin (already included within the containerized Fluentd instance) to send an additional copy of the captured messages outside of the framework. A separate instance of Fluentd must also be deployed in order to reveive messages sent by secure forward plugin. Once captured by the separate Fluentd instance, messages can then be sent to Splunk.
+
+image::images/logging-secure-forward-overview.png[title=Secure Forwarding to Splunk]
+
+The remainder of this document describes the process for implementing the integrated logging framework with Splunk.
+
+== Implementation
+
+This section describes how to configure the OpenShift's integrated logging solution to forward logs to splunk
+
+=== Prerequisites
+
+The following prerequisites are assumed and must be satisfied prior to implementing the solution
+
+* 2 dedicated machines
+** A instance of Splunk following the link:http://docs.splunk.com/Documentation/Splunk/latest/Installation/[installation documentation]
+** A machine for running a standalone instance of Fluentd
+* An OpenShift Container Platform cluster with aggregated logging installed
+
+=== Configure Splunk
+
+To accept messages sent from the fluent-plugin-splunk-ex plugin, a new TCP based data input must be configured within Splunk.
+
+As an Splunk user with administrative privileges, navigate to the Splunk user interface. Select *Settings* -> *Data Inputs* and then within the _Local Inputs_ section, select *Add new* next to the _TCP_ type. 
+
+Specify the following values to configure the TCP based Splunk input:
+
+* Port: 9997
+* Source name override: OCP
+* Source Type: _json
+
+NOTE: It is beyond the scope of this document for the creation of custom link:https://docs.splunk.com/Splexicon:Sourcetype[sourcetype's] to perform advanced parsing of the data received.
+
+Complete the necessary fields to finalize the addition of the new TCP input
+
+NOTE: Open applicable firewall ports on the Splunk machine to allow for data to be received 
+
+Since the fluent-plugin-splunk-ex plugin sends data to Splunk in batches, Splunk must be configured separate messages based on the presence of a new line character.
```

Comment:
add the word 'for'  after configured in this sentence.  example:

..., Splunk must be configured **for** separate messages based on...